src/electionguard-ui/ElectionGuard.UI/ViewModels/BaseViewModel.cs

```diff
@@ -34,12 +34,19 @@ private void InitTimer()
 
     public virtual async Task OnAppearing()
     {
+        UserName = AuthenticationService.UserName;
+        IsAdmin = AuthenticationService.IsAdmin;
+        SetPageTitle();
+
+        LocalizationService.OnLanguageChanged += OnLanguageChanged;
+
         _timer?.Start();
         await Task.Yield();
     }
 
     public virtual async Task OnLeavingPage()
     {
+        LocalizationService.OnLanguageChanged -= OnLanguageChanged;
```

Comment:
I don't think there's unit testing around this, did this change the page title to match the language we're localized into when on another page?

Reply:
Manually tested this with loading 2 different guardians and changing languages on the fly on the guardian home page after each logged in.  But we need to make sure that all of the other view models have the OnAppearing as well.